wdl/src/main/scala/wdl/Scatter.scala

```diff
@@ -68,6 +57,18 @@ object Scatter {
     }
     val possiblyNeededNestedOginPorts: Map[String, OutputPort] = possiblyNeededNestedOgins map { case (name: String, ogin: OuterGraphInputNode) => name -> ogin.singleOutputPort }
 
+    // Convert the scatter collection WdlExpression to a WdlWomExpression
+    val scatterCollectionExpression = WdlWomExpression(scatter.collection, scatter)
+    // Generate an ExpressionNode from the WdlWomExpression
+    val scatterCollectionExpressionNode =
+      WdlWomExpression.toAnonymousExpressionNode(WomIdentifier(scatter.item), scatterCollectionExpression, localLookup ++ possiblyNeededNestedOginPorts, Map.empty, preserveIndexForOuterLookups, scatter, PlainAnonymousExpressionNode.apply)
```

Comment:
This line isn't just a copy/paste. Note the `..., localLookup ++ possiblyNeededNestedOginPorts, Map.empty, ...`